compiler/src/dmd/dsymbolsem.d

```diff
@@ -7474,6 +7474,132 @@ private extern(C++) class NewScopeVisitor : Visitor
         sc = sc2;
     }
 }
+
+
+extern(C++) Dsymbols* include(Dsymbol d, Scope* sc)
+{
+    scope icv = new IncludeVisitor(sc);
+    d.accept(icv);
+    return icv.symbols;
+}
+
+extern(C++) class IncludeVisitor : Visitor
+{
+    alias visit = typeof(super).visit;
+    Scope* sc;
+    Dsymbols* symbols;
+    this(Scope* sc)
+    {
+        this.sc = sc;
+    }
+
+    override void visit(AttribDeclaration ad)
+    {
+        if (ad.errors)
+        {
+            symbols = null;
+            return;
+        }
+        symbols = ad.decl;
+        return;
+    }
+
+// Decide if 'then' or 'else' code should be included
+    override void visit(ConditionalDeclaration cdc)
+    {
+        //printf("ConditionalDeclaration::include(sc = %p) scope = %p\n", sc, _scope);
+
+        if (cdc.errors)
+        {
+            symbols = null;
+            return;
+        }
+        assert(cdc.condition);
+        symbols = cdc.condition.include(cdc._scope ? cdc._scope : sc) ? cdc.decl : cdc.elsedecl;
+    }
+
+    override void visit(StaticIfDeclaration sif)
+    {
+    /****************************************
+     * Different from other AttribDeclaration subclasses, include() call requires
+     * the completion of addMember and setScope phases.
+     */
+        //printf("StaticIfDeclaration::include(sc = %p) scope = %p\n", sc, _scope);
+        if (sif.errors || sif.onStack)
+        {
+            symbols = null;
+            return;
+        }
+        sif.onStack = true;
+        scope(exit) sif.onStack = false;
+
+        if (sc && sif.condition.inc == Include.notComputed)
+        {
+            assert(sif.scopesym); // addMember is already done
+            assert(sif._scope); // setScope is already done
+            Dsymbols* d = include(cast(ConditionalDeclaration)sif, sif._scope);
```

Comment:
This here doesn't work. The cast gets lost when calling include because include expects an AttribDeclaration and then it uses a visitor to get to the correct visiting override so you end up calling the StaticIfDeclaration override instead of the ConditionalDeclaration. The correct way to do this is:

```d
 Scope* saved_scope = sc;
  sc = sid._scope;
  visit(cast(ConditionalDeclaration) sif);
  Dsymbols* d = result;
  sc = saved_scope;

```

Reply:
Maybe something like this because `result` was uninitialized earlier @RazvanN7 @thewilsonator 
Scope* saved_scope = sc;  
 sc = sif._scope; 

Dsymbols* d = null;
visit(cast(ConditionalDeclaration) sif);
d = result;

sc = saved_scope;